elasticsearch/src/main/java/org/hibernate/ogm/datastore/elasticsearch/ElasticSearchDialect.java

```diff
@@ -0,0 +1,195 @@
+package org.hibernate.ogm.datastore.elasticsearch;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.action.get.GetResponse;
+import org.elasticsearch.action.index.IndexRequest;
+import org.elasticsearch.action.index.IndexResponse;
+import org.elasticsearch.action.search.SearchResponse;
+import org.elasticsearch.action.search.SearchType;
+import org.elasticsearch.common.annotations.VisibleForTesting;
+import org.elasticsearch.common.base.Function;
+import org.elasticsearch.common.collect.Lists;
+import org.elasticsearch.common.collect.Maps;
+import org.elasticsearch.common.settings.ImmutableSettings;
+import org.elasticsearch.index.query.QueryBuilders;
+import org.elasticsearch.search.SearchHit;
+import org.elasticsearch.search.SearchHitField;
+import org.hibernate.ogm.datastore.elasticsearch.impl.ElasticSearchDatastoreProvider;
+import org.hibernate.ogm.datastore.map.impl.MapHelpers;
+import org.hibernate.ogm.datastore.map.impl.MapTupleSnapshot;
+import org.hibernate.ogm.dialect.spi.AssociationContext;
+import org.hibernate.ogm.dialect.spi.AssociationTypeContext;
+import org.hibernate.ogm.dialect.spi.BaseGridDialect;
+import org.hibernate.ogm.dialect.spi.ModelConsumer;
+import org.hibernate.ogm.dialect.spi.NextValueRequest;
+import org.hibernate.ogm.dialect.spi.TupleAlreadyExistsException;
+import org.hibernate.ogm.dialect.spi.TupleContext;
+import org.hibernate.ogm.model.key.spi.AssociationKey;
+import org.hibernate.ogm.model.key.spi.AssociationKeyMetadata;
+import org.hibernate.ogm.model.key.spi.EntityKey;
+import org.hibernate.ogm.model.key.spi.EntityKeyMetadata;
+import org.hibernate.ogm.model.spi.Association;
+import org.hibernate.ogm.model.spi.Tuple;
+
+public class ElasticSearchDialect extends BaseGridDialect {
+
+	private final ElasticSearchDatastoreProvider provider;
+
+	private final Set<String> sequences = new HashSet<>();
+
+	public ElasticSearchDialect(ElasticSearchDatastoreProvider provider) {
+		super();
+		this.provider = provider;
+	}
+
+	@Override
+	public Tuple getTuple(EntityKey key, TupleContext tupleContext) {
+		GetResponse getResponse = provider.getClient().prepareGet(provider.getDatabase(), key.getTable(), id(key)).setOperationThreaded(false).execute().actionGet();
+		if (getResponse != null) {
+			return new Tuple(new MapTupleSnapshot(getResponse.getSource()));
+		}
+		return null;
+	}
+
+	private String id(EntityKey key) {
+		return String.valueOf(key.getColumnValues()[0]);
```

Comment:
It contains those column(s) representing the id property of the entity (annotated with `@Id`). There will be more than one column in case of a composite id type (see `@EmbeddedId`).